src/libraries/System.Private.CoreLib/src/System/IO/PathInternal.Windows.cs

```diff
@@ -341,6 +341,22 @@ internal static bool IsDirectorySeparator(char c)
             if (string.IsNullOrEmpty(path))
                 return path;
 
+            Span<char> destination = stackalloc char[path.Length];
```

Comment:
Unbounded stackallocs are prohibited in dotnet/runtime libraries

Reply:
@stephentoub I don't see record of a discussion about trying to catch this with an analyzer. It's not generally possible to be sure (either way) whether it's unbounded. But an analyzer that forced a pattern of passing a constant value would be doable.

Looking at all our stackallocs, almost all of them already pass a constant directly or via a constant field or local. The few exceptions are not self evidently safe by eyeball, eg.,
```c#
        private static unsafe int EncryptDecryptHelper(OP op, ISSPIInterface secModule, SafeDeleteContext context, Span<SecurityBuffer> input, uint sequenceNumber)
        {
            Interop.SspiCli.SecBufferDesc sdcInOut = new Interop.SspiCli.SecBufferDesc(input.Length);
            Span<Interop.SspiCli.SecBuffer> unmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[input.Length];
```
and it would be easy for one of them to be wrong. Would it be reasonable to have an analyzer that required them to be rewritten in constant terms, or at least something the analyzer could recognize like
```c#
            Span<Interop.SspiCli.SecBuffer> unmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[Math.Min(input.Length, BufferSize)];
```

Reply:
Moved to https://github.com/dotnet/runtime/issues/35382